astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -220,12 +220,15 @@ def validate(cls, value):
     ``transform_to()`` method of a coordinate frame or
     `~astropy.coordinates.SkyCoord` object::
 
+        >>> import warnings
         >>> import astropy.units as u
         >>> import astropy.coordinates as coord
         >>> c = coord.ICRS(ra=[158.3122, 24.5] * u.degree,
         ...                dec=[-17.3, 81.52] * u.degree,
         ...                distance=[11.5, 24.12] * u.kpc)
-        >>> c.transform_to(coord.Galactocentric) # doctest: +FLOAT_CMP
+        >>> with warnings.catch_warnings():
+        ...     warnings.simplefilter('ignore')  # Ignore deprecation warning
+        ...     c.transform_to(coord.Galactocentric) # doctest: +FLOAT_CMP
```

Comment:
I'm not a fan of needing to catch warnings in the doctests since this complicates the user examples. In fact if users run these examples we want them to see the warnings not ignore them. So I'm wondering if it would be better to include in the list of warnings we whitelist for tests in https://github.com/astropy/astropy/blob/master/astropy/tests/helper.py - alternatively we should set the value of ``galactocentric_frame_defaults`` at the start of the example. Same comment for the other changes in this file.

Reply:
I vaguely remember @adrn had a tough time setting the science state in doctest but I don't know if this particular doctest was affected by it.

Reply:
I agree with @astrofrog about catching warnings in the doctest! The fact that this issues a warning is actually "correct behavior", so maybe we want to just add words here to explain that it is expected?

Reply:
My goal is for `pytest` to not show that warning in the "warnings" when tests are done. I am not sure how explaining it in the doc will make `pytest` understand.

Reply:
Okay, let's see if `testsetup` is magical enough.

Reply:
@pllim - I guess that fundamentally what we present to the users is what matters the most, so we shouldn't obfuscate the examples just to avoid warnings in the test suite. I agree this example is tricky though, but we need to work around it in a way that doesn't complicate the user examples or they will copy that code and never see the warning they are supposed to see :)

Reply:
Yeah, I get it. I think the `testsetup` should be acceptable to us all?